services/worker/src/worker/job_runners/split/descriptive_statistics.py

```diff
@@ -578,7 +583,25 @@ def _compute_statistics(
         df_without_na = data.select(pl.col(column_name)).drop_nulls()
 
         lengths_column_name = f"{column_name}_len"
-        lengths_df = df_without_na.with_columns(pl.col(column_name).list.len().alias(lengths_column_name))
+        column_schema = data.schema[column_name]
+        if isinstance(column_schema, List):
+            lengths_df = df_without_na.with_columns(pl.col(column_name).list.len().alias(lengths_column_name))
+
+        elif isinstance(column_schema, Struct):
+            if not all(isinstance(field.dtype, List) for field in column_schema.fields):
+                raise PolarsSequenceSchemaError(
+                    f"Error parsing {column_name=} {column_schema=}. Subfields of Struct must be Lists. "
+                )
+            first_field_name = column_schema.fields[0].name  # assuming all fields have equal lengths within a sample
```

Comment:
to compute lengths in case when feature is dict of lists, i take values of the first field assuming that they all have different lengths. i don't know if it's possible that they actually have different lengths? cc @lhoestq

Reply:
They can have different lengths yes. For dicts I'm not sure what stats we should compute

Reply:
this specific case is for dicts of lists (it's just another option to represent list of structs) - so the same stats as for lists

Reply:
maybe just ignore structs for this PR ?

Reply:
like here in "emojis" and "labels" https://huggingface.co/datasets/OpenAssistant/oasst1?row=0

Reply:
>  maybe just ignore structs for this PR ?

this is the fix itself :smiley:

Reply:
I think the correct associated type is `struct` (not `sequence`) and for now we should not show anything for now. At least until we figure out a high level stat that makes sense for this type (maybe number of fields ?)

Reply:
OK: so, discriminating over the Sequence feature type is not a good idea because it can represent a list of things, or a dict of lists. We should show them as different types (here and/or on the Hub) to be sure to handle them differently, no?

Reply:
> in my opinion showing the histogram of length of the first list in the struct doesn't really make sense.

from what i see, this is just another way of storing the same data like this:
```python
[ {"name": "+1", "count": 10},  {"name": "_skip_reply", "count": 1},  {"name": "_skip_ranking", "count": 4}]
```
both ways in `datasets` are encoded as `Sequence({"name": Value("string"), "count": Value("int32")})` type; 
but in polars they are either `List` or `Struct(List)` depending on their raw values (if they are lists or dicts, as in example I provide here, or dicts of lists, as in your example above).

I can check if they have the same lengths, and return histogram only if it is true. But my point was that it is probably computationally inefficient while unnecessary, since we know that they are marked as `Sequence` so logically should be of the same lengths but technically there is no constraint for this. 
I can implement this, maybe at least compare if values of the first two keys have equal length in all rows.

>  the correct associated type is struct (not sequence) and for now we should not show anything for now

No, it is Sequence: https://huggingface.co/datasets/OpenAssistant/oasst1/blob/main/README.md?code=true#L58. I don't compute anything for struct for now indeed, only for sequences (you can check [_column_from_feature()](https://github.com/huggingface/datasets-server/pull/2602/files#diff-51b62bca3846b24d6554358a34bd39d0db4db7f24e378182fde2d549e2d1268bR756) function)

does this all make sense? :D i see it might me confusing

Reply:
It's named sequence for legacy reasons with tfds. Since the lists may not have the same length you can't really say it's equivalent to

```
[ {"name": "+1", "count": 10},  {"name": "_skip_reply", "count": 1},  {"name": "_skip_ranking", "count": 4}]
```

Reply:
I never really understood the Sequence type, but I assumed it's like a list, although it can be encoded in two ways (list of things, dict of lists). If it's the case (a list-like type), we should show the same stats as lists. If we're not sure it's always the case, we should maybe only process the "list" one, and ignore the "dict" one.

Reply:
> Since` the lists may not have the same length

@lhoestq do you think it makes sense to compare lengths of lists in first two fields and return stats if they are all equal?

Reply:
@severo we chatted with @lhoestq and for now we decided to disable sequence-dict case until we decide on what to do with nested things in the viewer.